### PR TITLE
fix(SankeyViz): enforce source/target order

### PR DESCRIPTION
**Pull request by serenajiang**
### SUMMARY
The SankeyViz assumes that the first column in the query object is the source and the second is the target. This changes unpredictably, so the SankeyViz sometimes switches randomly between the two options, even after a simple refresh.

This PR ensures that the `source` and `target` correspond to the first and second `groupby` columns.

### TEST PLAN
Test both configurations of source and target.

@etr2460 @ktmud ?

### CODE REVIEW

File: superset/viz.py [hunk 1]
```diff
@@ -1737,7 +1737,14 @@ def query_obj(self):
...
     def get_data(self, df: pd.DataFrame) -> VizData:
-        df.columns = ["source", "target", "value"]
+        df.rename(
+            columns={
+                self.form_data["groupby"][0]: "source",
+                self.form_data["groupby"][1]: "target",
+                df.columns[-1]: "value",
+            },
+            inplace=True,
+        )
```
Comment (ktmud) [inline superset/viz.py:1747]: How about

```python
        source, target = self.form_data["groupby"]
        metric = self.form_data["metric"]
        df.rename(
            columns={
                source: "source",
                target: "target",
                metric: "value",
            },
            inplace=True,
        )
```

Might want to do this for other network viz as well, such as `DirectedForceViz` and `ChordViz` below.

Comment (serenajiang) [reply superset/viz.py:1747]: `self.form_data["metric"]` does not seem to always be a string (e.g. if the metric is custom defined like `COUNT(col_1)`).

Comment (ktmud) [reply superset/viz.py:1747]: It seems the processed fields `self.groupby` and `self.metric_labels` might be more appropriate. Just wanted to be consistent how you get the original column names---either both from meta data, or both from the final df.

Comment (serenajiang) [reply superset/viz.py:1749]: @ktmud I altered it to use `self.groupby` and `self.metric_labels` (thanks for tip!)

I think I'll keep this PR specific to `SankeyViz`. I looked briefly into `ChordViz` and it looks like ChordViz uses a different metadata format, so this change might not be necessary.

**Review by ktmud — APPROVED** [review]

**Pull request merged**